test/extended/openstack/cloud-provider-config.go

```diff
@@ -66,55 +67,125 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] The Openshift", f
-						usr_property, _ := usr_section.GetKey(property_name)
-						int_property, _ := int_section.GetKey(property_name)
-						o.Expect(strings.ToLower(usr_property.Value())).To(o.Equal(strings.ToLower(int_property.Value())),
+							propertyName, sectionName, internalConfigMap.name, internalConfigMap.namespace)
+						usrProperty, _ := usrSection.GetKey(propertyName)
+						intProperty, _ := intSection.GetKey(propertyName)
+						o.Expect(strings.ToLower(usrProperty.Value())).To(o.Equal(strings.ToLower(intProperty.Value())),
 							"Unexpected value for property %q on section %q. configMap '%q', namespace '%q'.",
-							property_name, section_name, internalConfigMap.name, internalConfigMap.namespace)
+							propertyName, sectionName, internalConfigMap.name, internalConfigMap.namespace)
 						e2e.Logf("  - Property %q with correct value %q on section %q",
-							property_name, int_property.Value(), section_name)
+							propertyName, intProperty.Value(), sectionName)
 					}
 				}
 			}
 		})
+
+		//https://bugzilla.redhat.com/show_bug.cgi?id=2074471
+		g.It("should update use-octavia and enabled properties depending on the NetworkType", func() {
+
+			userConfigMap := configParams{
+				namespace: "openshift-config",
+				name:      "cloud-provider-config",
+				key:       "config",
+				skip:      nil,
+			}
+
+			ccmConfigMap := configParams{
+				namespace: "openshift-cloud-controller-manager",
+				name:      "cloud-conf",
+				key:       "cloud.conf",
+				skip:      nil,
+			}
+
+			sectionName := "LoadBalancer"
+			propertyNames := []string{"use-octavia", "enabled"}
+
+			cfg, err := e2e.LoadConfig()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			c, err := configv1client.NewForConfig(cfg)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			network, err := c.ConfigV1().Networks().Get(context.Background(), "cluster", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			userCfg, err := getConfig(clientSet,
+				userConfigMap.namespace,
+				userConfigMap.name,
+				userConfigMap.key)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			ccmCfg, err := getConfig(clientSet,
+				ccmConfigMap.namespace,
+				ccmConfigMap.name,
+				ccmConfigMap.key)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			var ccmPropertyValues []string
+			for _, propertyName := range propertyNames {
+				value, err := getPropertyValue(sectionName, propertyName, ccmCfg)
+				o.Expect(err).NotTo(o.HaveOccurred(), "Error getting the property %q in section %q on configMap "+
+					"%q (namespace: %q)", propertyName, sectionName, ccmConfigMap.name, ccmConfigMap.namespace)
+				ccmPropertyValues = append(ccmPropertyValues, value)
+			}
+
+			g.By(fmt.Sprintf("Checking properties for networkType detected: %q", network.Status.NetworkType))
+			var expectedValues []string
+			if network.Status.NetworkType == "Kuryr" {
+				//Fixed values regardless the content of userCfg.
+				expectedValues = []string{"true", "false"}
```

Comment:
Okay, so IIUC this only makes sense once we make a switch to use CCM. If so I would wait with merging this test until that happens as we're cautious with the possibility of it slipping again.

Reply:
In my local tests, I am not testing with external CCM enabled in the cluster, however the CCM configMap is configured. If you want to use this test to confirm that the fix implemented for https://bugzilla.redhat.com/show_bug.cgi?id=2074471 remains in place, I would merge it. But, if you think that behaviour could change, I am ok waiting until 4.12 enables CCM by default.

Reply:
Okay, I see the point, it's more about CCM bug even though CCM is just tech preview. Cool, I'm fine with this then, but I'll let @MaysaMacedo take a look too.